### PR TITLE
feat(cli): expose skill project name option

### LINKED ISSUE #1648: Expose --skill-project-name flag for local --skill-generate
When generating a skill locally with `--skill-generate`, the project name in `SKILL.md` (frontmatter `description:` and the H1) is always derived from the working-directory basename via `generateProjectName` → `toTitleCase(basename(cwd))`.

This is wrong when the cwd is an ephemeral/transient name. In our case the repo is checked out into git worktrees named per-instance (e.g. `karachi-v2`), so every generated skill leaks `Reference codebase for Karachi V2` instead of the actual project name.

The `skillProjectName` option already exists in the codebase (`packSkill.ts`: `const projectName = skillProjectName ?? generateProjectName(rootDirs)`) and `defaultAction.ts` reads it from `cliOptions` — but it is only ever populated on the **remote** path (`remoteAction.ts` → `generateProjectNameFromUrl`). There is no commander `.option()` for it and it is not in the config schema, so local users cannot set it.

### Proposal

Register a `--skill-project-name <name>` CLI option (and/or an `output.skillProjectName` config field) that flows into the existing `skillProjectName` plumbing. Since the wiring already exists, this should be a small change.

### Workaround

Post-process `SKILL.md` + `references/*.md` after generation to replace the derived name — works, but brittle and should not be necessary given the field already exists internally.

Tested with `repomix@^1.14.1`.

### PR DESCRIPTION
**Pull request by WilliamK112**
## Summary
- Add `--skill-project-name <name>` to local `--skill-generate` CLI usage.
- Validate that the option is only used with skill generation and is not empty.
- Document the new flag and cover option propagation/validation in tests.

Closes #1648

## Validation
- `npm run test -- --run`
- `npx biome check .`
- `npx oxlint .`
- `npm run lint-ts`
- `npx secretlint **/* --secretlintignore .gitignore`
- `npm run build`
- `node bin\repomix.cjs --help | Select-String -Pattern skill-project-name|skill-generate|skill-output`

## Checklist

- [x] Run `npm run test`
- [x] Run `npm run lint`

### CODE REVIEW

File: src/cli/actions/defaultAction.ts [hunk 1]
```diff
@@ -62,11 +62,17 @@ export const runDefaultAction = async (
   if (cliOptions.force && config.skillGenerate === undefined) {
     throw new RepomixError('--force can only be used with --skill-generate');
   }
+  if (cliOptions.skillProjectName && config.skillGenerate === undefined) {
+    throw new RepomixError('--skill-project-name can only be used with --skill-generate');
+  }
```
[the remark this answers is not in the document]
Comment (WilliamK112) [reply src/cli/actions/defaultAction.ts]: Addressed in ccb3e6f by switching the guard to `skillProjectName !== undefined` and adding a regression test for an empty value without `--skill-generate`.

**Review by yamadashy — APPROVED** [review]
Thanks for the clean, well-tested PR @WilliamK112! 🎉

This nicely exposes the existing `skillProjectName` plumbing for local use, and the two guards follow the same style as the sibling `--skill-output` / `--force` validations.

**Comment by yamadashy** [comment]
I'll merge this now. Thanks again! 🎉
